### PR TITLE
Add TLS to pico-w5500 baremetal example

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

**Review by scaprile — CHANGES_REQUESTED** [review]

**Comment by scaprile** [comment]
Some testing mods seem to have slipped into the commit.
If it is for the F746, I'll take over the fight, otherwise, let me know

File: mip/mip.c [hunk 1]
```diff
@@ -609,9 +609,14 @@ static void read_conn(struct mg_connection *c, struct pkt *pkt) {
...
-    // than close this connection
-    mg_error(c, "SEQ != ACK: %x %x", seq, s->ack);
+    uint32_t ack = (uint32_t) (mg_htonl(pkt->tcp->seq) + pkt->pay.len);
+    if (s->ack == ack) {
+      MG_VERBOSE(("ignoring duplicate pkt"));
+    } else {
+      // TODO(cpq): peer sent us SEQ which we don't expect. Retransmit rather
+      // than close this connection
+      mg_error(c, "SEQ != ACK: %x %x %x", seq, s->ack, ack);
+    }
```
[the remark this answers is not in the document]
Comment (cpq) [reply mip/mip.c:619]: Without this commit, `curl -k https://IP` does not work! So it is essential.

File: mip/driver_stm32.c [hunk 2]
```diff
@@ -179,9 +180,11 @@ static bool mip_driver_stm32_up(struct mip_if *ifp) {
   return bsr & BIT(2) ? 1 : 0;
 }
 
-void ETH_IRQHandler(void);
 static uint32_t s_rxno;
+void ETH_IRQHandler(void);
 void ETH_IRQHandler(void) {
+  static int cnt;
+  printf("--> %d\n", cnt++);
```
[the remark this answers is not in the document]
Comment (cpq) [reply mip/driver_stm32.c:187]: Oops.. F746 leftovers

File: mip/driver_stm32.c [hunk 3]
```diff
@@ -166,6 +166,7 @@ static size_t mip_driver_stm32_tx(const void *buf, size_t len, struct mip_if *if
     s_txdesc[s_txno][0] = BIT(20) | BIT(28) | BIT(29) | BIT(30);  // Chain,FS,LS
     s_txdesc[s_txno][0] |= BIT(31);  // Set OWN bit - let DMA take over
     if (++s_txno >= ETH_DESC_CNT) s_txno = 0;
+    printf("sent\n");
```
[the remark this answers is not in the document]
Comment (cpq) [reply mip/driver_stm32.c:169]: Oops.. F746 leftovers!

**Comment by cpq** [comment]
Yeah, ST changes should not be there, let me clean them up...

File: mip/mip.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (scaprile) [reply mip/mip.c:619]: I meant it belongs to a separate MIP fix/enhancement commit. Let's push it anyway and I'll check later. I wonder why a duplicated ACK/segment made its way, but that is not a reason to close a connection.

**Review by scaprile — APPROVED** [review]

**Pull request merged**